### PR TITLE
#1319 Summarizing user for the back API response

### PR DESCRIPTION
**Pull request by zabil**
This also fixes problem with serialization when go server is used without authentication.

### CODE REVIEW

File: server/webapp/WEB-INF/rails.new/app/presenters/api_v1/backup_representer.rb [hunk 1]
```diff
@@ -16,13 +16,11 @@
...
 module ApiV1
   class BackupRepresenter < ApiV1::BaseRepresenter
-    attr_reader :user
     alias_method :backup, :represented
 
-    def initialize(backup_and_user)
-      backup, user = *backup_and_user
+    def initialize(backup)
+      backup
       super(backup)
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/app/presenters/api_v1/backup_representer.rb]: @zabil - Don't need the constructor, in this case

**Comment by ketan** [comment]
In the world of hypermedia - I'm curious why this was removed? If you're rendering information about a different type of resource, why not render the resource (or a url to that resource).

An API without hyperlinks is not where we want to go again. Ever.

**Comment by zabil** [comment]
Not sure if I understand.

Why should the backup API return the entire user representation (with email, aliases etc) a hyperlink to the user resource makes sense but when go is used without auth there's no user resource how do we handle that?

**Comment by ketan** [comment]
> Why should the backup API return the entire user representation (with email, aliases etc) a hyperlink to the user resource makes sense but when go is used without auth there's no user resource how do we handle that?

That is a certainly a problem to fix - but I don't think removing the user resource for everyone is a solution for that problem.

**Comment by zabil** [comment]
>  but I don't think removing the user resource for everyone is a solution for that problem.
OK

Here's an attempt to fix that

Is changing  
```
{
  "_links": {
    "doc": {
      "href": "http://api.go.cd/#backups"
    }
  },
  "time": "2015-07-01T14:21:46.593+05:30",
  "path": "/var/lib/go-server/serverBackups/backup_20150701-142146",
  "user": {
    "_links": {
      "doc": {
        "href": "http://api.go.cd/#users"
      },
      "self": {
        "href": "https://ci.example.com/go/api/users/admin"
      },
      "find": {
        "href": "https://ci.example.com/go/api/users/:login_name"
      }
    },
    "login_name": "admin",
    "display_name": "admin",
    "enabled": true,
    "email": null,
    "email_me": false,
    "checkin_aliases": [

    ]
  }
```

to 

```
{
  "_links": {
    "doc": {
      "href": "http://api.go.cd/#backups"
    }
  },
  "time": "2015-07-01T14:21:46.593+05:30",
  "path": "/var/lib/go-server/serverBackups/backup_20150701-142146",
  "user": {
    "_links": {
      "doc": {
        "href": "http://api.go.cd/#users"
      },
      "self": {
        "href": "https://ci.example.com/go/api/users/admin"
      },
      "find": {
        "href": "https://ci.example.com/go/api/users/:login_name"
      }
    }
  }
```

right?

And what to do with user when unauthenticated?

**Comment by ketan** [comment]
That sounds reasonable. Rendering a link to the user object, without rendering the user attributes sounds good.

**Pull request merged**